BOOTSEQ.md

```diff
@@ -4,77 +4,160 @@
+
+- xPU starts booting
+- xPU completes boot and issues hot-plug event
+- Host updates root complex with xPU resources and makes those resources available to services executing on the host
+- _Note: This assumes host will properly execute a hot-plug event_
 - More...
 
-## xPU and host reboot
+## Host OS Reset or Crash
+
+- During host OS Reboot or Crashes, the host writes to Chipset Reset Control Register (i.e., IO port 0xCF9) or to the ACPI FADT
+- The Platform (CPLD) firmware can monitor for these events and trigger host BMC/BIOS to coordinate with xPU to take appropriate action. Potential actions include:-
+  - Ensuring that xPU is gracefully shut down before a full host reboot
+  - Force reset xPU after a timeout if xPU has failed to gracefully shutdown
+  - Allow xPU to continue executing if there is no host/xPU dependency
+
+## xPU OS Reset or Crash
+
+xPU OS/FW crash and/or independent reboot will result in a Downstream Port Containment (DPC) event on the PCIe bus
+
+- The host (BMC/BIOS) firmware and host OS needs to implement DPC and hot plug requirements defined in PCIe Firmware spec 3.3 for these scenarios to work.
 
-- xPU OS/FW reboot assumed to cause Host OS crash
-  - Not in all cases. e.g. NVIDIA's design separates the various components, and an ARM complex reboot should have no direct effect on the host OS. Similarly, some vendors implement ISSU processes for certain FW updates where no disruption is observed.
+- Coordinated shutdown
+  - The xPU can provide in-band or out-of band signalling to host OS that it intends to restart. The host OS can respond by either
+    - Ignoring the event if the host OS intends to continue execution after the remove event, containing the subsequent surprise removal
+    - Host OS reset (see above), to coordinate reboot of other xPUs on the host bus
+
+- xPU OS crash or reset.
+  - xPU will trigger DPC/CI surprise remove event.
+  - Host OS and BMC/BIOS will need to handle these events gracefully so that host OS remains active for long enough for the xPU OS crash dump to complete.
+  - If host OS is tightly coupled to xPU function, it can elect to gracefully shutdown/reboot when the xPU crashdump has completed.
+  - Alternatively, if host OS is losely coupled or decoupled from xPU (if xPU is acting as a bump in the wire service for example), the host can
+    - Elect to contain the event, removing the xPU from the bus table. When the xPU restarts, it will generate a hot-plug event on the PCIe bus, allowing the host BIOS/OS to reinsert the xPU.
+    - Note: This behavior is currently a largely untested feature in Linux host OS, and is not available on Windows
+
+  - _Further discussion required_ NVIDIA's design separates the various components, and an ARM complex reboot should have no direct effect on the host OS. Similarly, some vendors implement ISSU processes for certain FW updates where no disruption is observed.
```

Comment:
Ack - it depends on the what you are upgrading/changing. But why ISSU is kept under xPU Crash/reset bullet?